### PR TITLE
feat: add ResolveResponse::to_result()

### PR DESCRIPTION
**Pull request by kitsonk**
Found a pattern that should be in deno_graph when refactoring.

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM

File: src/source.rs [hunk 1]
```diff
@@ -112,6 +112,21 @@ pub enum ResolveResponse {
   Umd(ModuleSpecifier),
 }
 
+impl ResolveResponse {
+  pub fn to_result(self) -> Result<ModuleSpecifier, Error> {
```
Comment (dsherret) [inline src/source.rs:116]: Maybe `to_specifier` instead?

**Review by dsherret — APPROVED** [review]

File: src/source.rs (hunk 1 above)
Comment (kitsonk) [reply src/source.rs:116]: Hmmm... I would prefer result, because it unwraps the enum to a result.

Comment (dsherret) [reply src/source.rs:116]: What about `try_specifier()` or something like that?

**Pull request merged**